cmake/macros/macro_deal_ii_setup_target.cmake

```diff
@@ -21,21 +21,9 @@
 #       deal_ii_setup_target(target)
 #       deal_ii_setup_target(target DEBUG|RELEASE)
 #
-# This appends necessary include directories, linker flags, compile flags
-# and compile definitions and the deal.II library link interface to the
-# given target. In particular:
-#
-# INCLUDE_DIRECTORIES is appended with
-#   "${DEAL_II_INCLUDE_DIRS}"
-#
-# COMPILE_FLAGS is appended with
-#   "${DEAL_II_CXX_FLAGS} ${DEAL_II_CXX_FLAGS_<build type>}"
-#
-# LINK_FLAGS is appended with
-#   "${DEAL_II_LINKER_FLAGS ${DEAL_II_LINKER_FLAGS_<build type>}"
-#
-# COMPILE_DEFINITIONS is appended with
-#   "${DEAL_II_DEFINITIONS};${DEAL_II_DEFINITIONS_<build type>}"
+# This appends the deal.II target to the link interface of the specified
+# target, which in turn ensures that necessary include directories, linker
+# flags, compile flags and compile definitions are set.
```

Comment:
And we want people to migrate to `target_link_libraries(my_target ${DEAL_II_TARGET})` instead, right?

Reply:
What the `deal_ii_setup_target()` macro does is to
 - check whether a Release or Debug mode is set and error out if not
 - do this `target_link_libraries()` call
 - but also populate compiler and linker options.